### ISSUE TITLE
What's difference between PyTorch:Softmax & openfold/utils/kernel/csrc/softmax

### ISSUE THREAD

**Issue by jinmingyi1998** [aqlaboratory/openfold#139]
they both calculate softmax on cuda, process similar operations. so why dont we import torch.nn.Softmax

**Comment by gahdritz** [comment]
torch.nn.softmax consumes massive amounts of memory. See https://github.com/pytorch/pytorch/issues/72450. Our custom CUDA kernel uses five times less.
